src/ray/raylet/scheduling/local_resource_manager.cc

```diff
@@ -306,19 +306,25 @@ void LocalResourceManager::UpdateAvailableObjectStoreMemResource() {
   RAY_CHECK_EQ(total_instances.size(), 1u);
   const double used = get_used_object_store_memory_();
   const double total = total_instances[0].Double();
-  local_resources_.available.Set(ResourceID::ObjectStoreMemory(),
-                                 {FixedPoint(total >= used ? total - used : 0.0)});
-
-  OnResourceChanged();
+  auto new_available =
+      std::vector<FixedPoint>{FixedPoint(total >= used ? total - used : 0.0)};
+  if (new_available != local_resources_.available.Get(ResourceID::ObjectStoreMemory())) {
+    local_resources_.available.Set(ResourceID::ObjectStoreMemory(),
+                                   std::move(new_available));
+    OnResourceChanged();
+  }
 }
 
-void LocalResourceManager::FillResourceUsage(rpc::ResourcesData &resources_data) {
-  UpdateAvailableObjectStoreMemResource();
+void LocalResourceManager::FillResourceUsage(rpc::ResourcesData &resources_data,
+                                             bool for_ray_syncer) {
+  if (!for_ray_syncer) {
```

Comment:
hmm why for_ray_syncer is different?

Reply:
Yeah +1 to this quesiton!

Reply:
@rkooo567 @scv119  
The reason is that, the previous ones, do the diff comparison to see whether we need to send the message or not.

The new ones doing the thing: check the version number.


Also, the previous ones separate the fields into even smaller ones but the new ones, treat it as a whole snapshot.

I think maybe a better way is to have another function for this one.